### PR TITLE
Split ImageData: extract HDF5 store, remove active_image, add ImageKey

### PR DESCRIPTION
**Pull request by sakoho81**
## Summary

Refactors the 774-line `ImageData` class into a composable architecture:

### New

- **`ImageType`** (StrEnum) — type-safe replacement for `image_types_c`
- **`ImageKey`** (frozen dataclass) — bundles (image_type, index, channel, scale) into one parameter
- **`ImageDataStore`** (Protocol) — storage interface for multi-view image data
- **`HDF5ImageStore`** — HDF5 implementation of the protocol
- **`image_data_operations.py`** — standalone functions for `create_rescaled_images`, `calculate_missing_psfs`, `copy_registration_result`

### Changed

- **`ImageData`** — now a thin facade (~180 lines, down from 774) composing `HDF5ImageStore`. All getters take explicit `ImageKey`. Removed: `active_image`, `set_active_image`, `__getitem__`, `__setitem__`, `print()`, `uiutils.get_user_input` from data layer.
- **8 consumer files** updated to use `ImageKey`-based API
- **`definitions.py`** — removed `image_types_c`

### API before/after

```python
# Before: 2-step stateful
data.set_active_image(view, ch, sc, "registered")
val = data.get_max()

# After: 1-step stateless with reusable key
key = ImageKey(ImageType.REGISTERED, view, ch, sc)
val = data.get_max(key)
```

### Tests

- 44 new tests for store, protocol, facade, path construction, edge cases
- 339 total passing (295 original + 44 new)
- Ruff + mypy clean

### CODE REVIEW

File: miplib/bin/import.py [hunk 1]
```diff
@@ -91,7 +92,7 @@ def main():
             images = (images * (255.0 / images.max())).astype(numpy.uint8)
 
         if not all(x in image_name for x in params_c) or not any(
-            x in image_name for x in image_types_c
+            x in image_name for x in ImageType
         ):
             print(f"Unrecognized image name {image_name}. Skipping it.")
```
Comment (sakoho81) [inline miplib/bin/import.py]: the printout is a bit weird here. I suppose we want to say that image of unknown type?

File: miplib/data/containers/image_data_store.py [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+        channel: int,
+        scale: int,
+        params: Sequence[float],
+        fixed_params: Sequence[float],
+        transform_type: int,
+    ) -> None: ...
+
+    def get_image_data(self, key: ImageKey) -> np.ndarray: ...
+
+    def get_image_attributes(self, key: ImageKey) -> dict[str, Any]: ...
```
Comment (sakoho81) [inline miplib/data/containers/image_data_store.py]: lets add basic docstrings to document what is the intent of the methods in the interface

File: miplib/data/image_data_operations.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+from miplib.data.containers.image_data_store import (
+    ImageDataStore,
+    ImageKey,
+    ImageType,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def create_rescaled_images(
```
Comment (sakoho81) [inline miplib/data/image_data_operations.py:16]: also this file.  Lets add docstrings to the functions

File: miplib/processing/fusion/utils.py [hunk 4]
```diff
@@ -1,49 +1,57 @@
...
 
 
-def sum_of_all(data_structure, channel=0, scale=100, image_type="original"):
+def _to_imagetype(image_type: ImageType | str) -> ImageType:
+    if isinstance(image_type, ImageType):
+        return image_type
+    return ImageType(image_type)
+
+
+def sum_of_all(data_structure, channel=0, scale=100, image_type=ImageType.ORIGINAL):
```
Comment (sakoho81) [inline miplib/processing/fusion/utils.py]: also here, docstrings, aslso type annotations are quite lacking
